### ISSUE TITLE
Backspace doesn't work correctly if `input_type` is set to "number"

### ISSUE THREAD

**Issue by DexerBR** [kivy/kivy#8232]
**Software Versions**
* Python: `3.9.9` / `3.8.9`
* OS: Ubuntu 20.04 (WSL2)
* Kivy: `2.1.0` /  `master`
* Kivy installation method: through buildozer (`kivy==2.1.0`, `kivy==master`)
* Buildozer: `1.5.0`
* python-for-android: `v2023.02.10`
* Device: Amazon Fire Tablet 8 10th gen

**Describe the bug**

When setting the `input_type` to `number` ( in the described environment), the backspace stops working correctly if there is a initial text. It seems that the starting character count is stuck, and in some cases this stuck character count seems to increase or decrease.

I also tested the `datetime` and `tel` options, and the behavior is the same.

**Expected behavior**
Backspace should work correctly

**To Reproduce**

```python
from kivy.app import App
from kivy.lang import Builder


KV = """
BoxLayout:
    TextInput:
        text: "12"
        input_type: "number"
"""


class TestApp(App):
    def build(self):
        return Builder.load_string(KV)


if __name__ == "__main__":
    TestApp().run()
```

**Code and Logs and screenshots**

https://user-images.githubusercontent.com/73297572/236015364-fee3d6f4-bbf7-46c7-b17e-ae4b89894df3.mp4

**Comment by misl6** [comment]
Yeah, I guess we can add this issue to  https://github.com/kivy/kivy/pull/8226 "related issues" long list.

**Comment by afriddev** [comment]
Upgrade The version
